compiler/src/dotty/tools/repl/ReplDriver.scala

```diff
@@ -517,4 +543,90 @@ class ReplDriver(settings: Array[String],
     case interfaces.Diagnostic.INFO => out.println(dia.msg) // print REPL's special info diagnostics directly to out
     case _                          => ReplConsoleReporter.doReport(dia)(using state.context)
 
+  extension (sym: Symbol)(using Context)
+    // borrowed from ExtractDependencies#recordDependency and adapted to handle @targetName
+    private def binaryClassName: String =
```

Comment:
To expand on Seth's comment, I believe this PR correctly deals with classes with the `@targetName` attribute because I borrowed and adapted this `binaryClassName` code from `ExtractDependencies`.  But I suspect there may be incremental compilation issues related to `@targetName`, and ultimately I think this `binaryClassName` code should be shared rather than duplicated.

I'm going to try to find some test cases and open a separate ticket for any incremental compilation issues.